org/mozilla/jss/pkcs11/PK11PrivKey.c

```diff
@@ -665,3 +665,64 @@ Java_org_mozilla_jss_pkcs11_PK11PrivKey_getDSAParamsNative
 
     return pqgArray;
 }
+
+/**********************************************************************
+ * PK11RSAPrivateKey.getModulusByteArray
+ */
+JNIEXPORT jbyteArray JNICALL
+Java_org_mozilla_jss_pkcs11_PK11RSAPrivateKey_getModulusByteArray
+    (JNIEnv *env, jobject this)
+{
+    SECKEYPrivateKey *privateKey;
+    SECKEYPublicKey *publicKey;
+    PK11SlotInfo *slot;
+    int rv;
+    jbyte *value;
+    jint length;
+    jbyteArray array = NULL;
+
+    PR_ASSERT(env!=NULL && this!=NULL);
+
+    // get private key
+    rv = JSS_PK11_getPrivKeyPtr(env, this, &privateKey);
+
+    // check return code
+    if (rv != PR_SUCCESS) {
+        char *message = PR_smprintf("Unable to get RSA private key (rc: %d)", rv);
+        JSS_throwMsg(env, PK11_EXCEPTION, message);
+        PR_smprintf_free(message);
+        goto finish;
+    }
+
+    // get slot from private key
+    slot = PK11_GetSlotFromPrivateKey(privateKey);
```

Comment:
I think you need to free that slot object: See here: https://dxr.mozilla.org/security/source/mozilla/security/nss/lib/ssl/ssl3con.c#5359